src/gsy_e/models/area/scm_manager.py

```diff
@@ -441,7 +465,7 @@ def _validate_coefficients(community: "CoefficientArea") -> None:
                 f"Home {home.name} has assets with non-SCM strategies."
 
     @staticmethod
-    def _validate_market_maker_rate(community):
+    def _validate_market_maker_rate(community: "CoefficientArea") -> None:
         for home in community.children:
-            assert home.get("market_maker_rate") is not None, \
+            assert home._market_maker_rate is not None, \
```

Comment:
Just out of curiosity, how it is guaranteed that _market_maker_rate is always an attribute of each home?

Reply:
The type hint is supposed to protect us from that. It explicitly defines that a CoefficientArea is passed in this method, and the _market_maker_rate is explicitly defined in the constructor of the CoefficientArea class. 
We can make it even more ironclad by adding an assertion that the area class passed in this method is indeed a CoefficientArea, however I thought that it was an overkill while implementing it. If you insist though I will add it.